src/poetry/console/commands/init.py

```diff
@@ -145,13 +145,11 @@ def handle(self) -> int:
         python = self.option("python")
         if not python:
             current_env = SystemEnv(Path(sys.executable))
-            default_python = "^{}".format(
-                ".".join(str(v) for v in current_env.version_info[:2])
+            default_python = "^" + ".".join(
```

Comment:
in some places, it was easier to use `Version.from_parts(*sys.version_info[:3])`. i experimented with the helper below, but wanted to explore if it made more sense to use the `Version` class everywhere. note i've left the helper out of this pr to avoid delaying it longer:

```python
def fmt_version(
    version: Tuple[int, int, int],
    precision: Literal["major", "minor", "patch"] = "patch",
) -> str:
    major, minor, patch, *_ = version
    if precision == "major":
        return f"{major}"
    elif precision == "minor":
        return f"{major}.{minor}"
    else:
        return f"{major}.{minor}.{patch}"
```